### ISSUE TITLE
Illegal Memory Access with ShuffleNet

### ISSUE THREAD

**Issue by twmht** [NVIDIA/TensorRT#780]
## Environment

**TensorRT Version**:  7.1.3
**GPU Type**:  GTX-1070-Ti
**Nvidia Driver Version**:  440.33.01
**CUDA Version**:  10.2
**CUDNN Version**:  8.0.2
**Operating System + Version**:  ubuntu 18.04
**Python Version (if applicable)**:  3.7.4
**TensorFlow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 
**Baremetal or Container (which commit + image + tag)**: 

## Relevant Files
onnx model: [shufflenet_v2_retinaface.zip](https://github.com/NVIDIA/TensorRT/files/5223614/shufflenet_v2_retinaface.zip)

## Steps To Reproduce

```python
with open(onnx_path, 'rb') as f:
    onnx_bytes = f.read()
height = 640
width = 640
log_level=trt.Logger.VERBOSE
logger = trt.Logger(log_level)
builder = trt.Builder(logger)
network = builder.create_network(1 << int(trt.NetworkDefinitionCreationFlag.EXPLICIT_BATCH))
parser = trt.OnnxParser(network, logger)
parser.parse(onnx_bytes)
config.max_workspace_size = 1<<25
config.min_timing_iterations = 3
config.avg_timing_iterations = 2
max_batch_size = 5
profile = builder.create_optimization_profile()
#  profile.set_shape('input.1', min=(1, 3, 480, 640), opt=(3, 3, 480, 640), max=(max_batch_size, 3, 480, 640))
#  profile.set_shape('input.1', min=(1, 3, 640, 480), opt=(3, 3, 640, 480), max=(max_batch_size, 3, 640, 480))
profile.set_shape('input', min=(1, 3, height, width), opt=(3, 3, height, width), max=(max_batch_size, 3, height, width))
config.add_optimization_profile(profile)
builder.max_batch_size = max_batch_size
engine = builder.build_engine(network, config)
```

the errors are

```
[TensorRT] VERBOSE: --------------- Timing Runner: Reshape_19 + Transpose_20 (Shuffle)
[TensorRT] ERROR: ../builder/cudnnBuilderUtils.cpp (427) - Cuda Error in findFastestTactic: 700 (an illegal memory access was encountered)
[TensorRT] ERROR: ../rtSafe/safeRuntime.cpp (32) - Cuda Error in free: 700 (an illegal memory access was encountered)
terminate called after throwing an instance of 'nvinfer1::CudaError'
  what():  std::exception
```

any idea?

**Comment by twmht** [comment]
it's related to shufflenetV2, onnx models in https://github.com/onnx/models/tree/master/vision/classification/shufflenet also have this issue.

**Comment by zerollzeng** [comment]
I also meet this problem while testing dynamic shapes on shuffle net 
## step to reproduce
1. download the onnx model from https://github.com/onnx/models/blob/master/vision/classification/shufflenet/model/shufflenet-v2-10.onnx
2. run convert code to unset fixed shape
```python
import sys
import onnx
model_path = str(sys.argv[1])
model = onnx.load(model_path)
model.graph.input[0].type.tensor_type.shape.dim[0].dim_param = '?'
# model.graph.input[0].type.tensor_type.shape.dim[1].dim_param = '?'
model.graph.input[0].type.tensor_type.shape.dim[2].dim_param = '?'
model.graph.input[0].type.tensor_type.shape.dim[3].dim_param = '?'
onnx.save(model, 'dynamic_model.onnx')
```
python onnx.model shufflenet-v2-10.onnx
3. /usr/src/tensorrt/bin/trtexec --explicitBatch --onnx=../models/dynamic_model.onnx --minShapes=input:1x3x10x10 --optShapes=input:1x3x1024x1024 --maxShapes=input:1x3x1224x1224
and the output
```
[10/22/2020-16:33:58] [I] === Model Options ===                                                                                                                                                                                             
[10/22/2020-16:33:58] [I] Format: ONNX                                                                                                                                                                                                      
[10/22/2020-16:33:58] [I] Model: ../models/dynamic_model.onnx                                                                                                                                                                               
[10/22/2020-16:33:58] [I] Output:                                                                                                                                                                                                           
[10/22/2020-16:33:58] [I] === Build Options ===                                                                                                                                                                                             
[10/22/2020-16:33:58] [I] Max batch: explicit                                                                                                                                                                                               
[10/22/2020-16:33:58] [I] Workspace: 16 MB                                                                                                                                                                                                  
[10/22/2020-16:33:58] [I] minTiming: 1                                                                                                                                                                                                      
[10/22/2020-16:33:58] [I] avgTiming: 8                                                                                                                                                                                                      
[10/22/2020-16:33:58] [I] Precision: FP32                                                                                                                                                                                                   
[10/22/2020-16:33:58] [I] Calibration:                                                                                                                                                                                                      
[10/22/2020-16:33:58] [I] Safe mode: Disabled                                                                                                                                                                                               
[10/22/2020-16:33:58] [I] Save engine:                                                                                                                                                                                                      
[10/22/2020-16:33:58] [I] Load engine:                                                                                                                                                                                                      
[10/22/2020-16:33:58] [I] Inputs format: fp32:CHW                                                                                                                                                                                           
[10/22/2020-16:33:58] [I] Outputs format: fp32:CHW                                                                                                                                                                                          
[10/22/2020-16:33:58] [I] Input build shape: input=1x3x10x10+1x3x1024x1024+1x3x1224x1224                                                                                                                                                    
[10/22/2020-16:33:58] [I] === System Options ===                                                                                                                                                                                            
[10/22/2020-16:33:58] [I] Device: 0                                                                                                                                                                                                         
[10/22/2020-16:33:58] [I] DLACore:                                                                                                                                                                                                          
[10/22/2020-16:33:58] [I] Plugins:                                                                                                                                                                                                          
[10/22/2020-16:33:58] [I] === Inference Options ===                                                                                                                                                                                         
[10/22/2020-16:33:58] [I] Batch: Explicit                                                                                                                                                                                                   
[10/22/2020-16:33:58] [I] Iterations: 10                                                                                                                                                                                                    
[10/22/2020-16:33:58] [I] Duration: 3s (+ 200ms warm up)                                                                                                                                                                                    
[10/22/2020-16:33:58] [I] Sleep time: 0ms                                                                                                                                                                                                   
[10/22/2020-16:33:58] [I] Streams: 1                                                                                                                                                                                                        
[10/22/2020-16:33:58] [I] ExposeDMA: Disabled                                                                                                                                                                                               
[10/22/2020-16:33:58] [I] Spin-wait: Disabled                                                                                                                                                                                               
[10/22/2020-16:33:58] [I] Multithreading: Disabled                                                                                                                                                                                          
[10/22/2020-16:33:58] [I] CUDA Graph: Disabled                                                                                                                                                                                              
[10/22/2020-16:33:58] [I] Skip inference: Disabled                                                                                                                                                                                          
[10/22/2020-16:33:58] [I] Inputs:                                                                                                                                                                                                           
[10/22/2020-16:33:58] [I] === Reporting Options ===                                                                                                                                                                                         
[10/22/2020-16:33:58] [I] Verbose: Disabled                                                                                                                                                                                                 
[10/22/2020-16:33:58] [I] Averages: 10 inferences                                                                                                                                                                                           
[10/22/2020-16:33:58] [I] Percentile: 99
[10/22/2020-16:33:58] [I] Dump output: Disabled
[10/22/2020-16:33:58] [I] Profile: Disabled
[10/22/2020-16:33:58] [I] Export timing to JSON file: 
[10/22/2020-16:33:58] [I] Export output to JSON file: 
[10/22/2020-16:33:58] [I] Export profile to JSON file: 
[10/22/2020-16:33:58] [I] 
----------------------------------------------------------------
Input filename:   ../models/dynamic_model.onnx
ONNX IR version:  0.0.6
Opset version:    10
Producer name:    pytorch
Producer version: 1.6
Domain:           
Model version:    0
Doc string:       
----------------------------------------------------------------
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] onnx2trt_utils.cpp:198: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/22/2020-16:34:00] [W] [TRT] Calling isShapeTensor before the entire network is constructed may result in an inaccurate result.
[10/22/2020-16:34:01] [I] [TRT] Some tactics do not have sufficient workspace memory to run. Increasing workspace size may increase performance, please check verbose output.
[10/22/2020-16:34:03] [E] [TRT] ../builder/cudnnBuilderUtils.cpp (372) - Cuda Error in findFastestTactic: 77 (an illegal memory access was encountered)
[10/22/2020-16:34:03] [E] [TRT] ../rtSafe/safeRuntime.cpp (32) - Cuda Error in free: 77 (an illegal memory access was encountered)
terminate called after throwing an instance of 'nvinfer1::CudaError'
  what():  std::exception
Aborted (core dumped)
```

**Comment by twmht** [comment]
@mk-nvidia 

how to set such layers to dynamic?

**Comment by aojue1109** [comment]
how to set such layers to dynamic?
